diesel/src/macros/mod.rs

```diff
@@ -26,47 +26,31 @@ macro_rules! __diesel_column {
         impl SelectableExpression<$($table)::*> for $column_name {
         }
 
-        impl AppearsOnTable<$($table)::*> for $column_name {
-        }
-
-        impl<Right, Kind> SelectableExpression<
-            Join<$($table)::*, Right, Kind>,
-        > for $column_name where
-            $column_name: AppearsOnTable<Join<$($table)::*, Right, Kind>>,
-        {
-        }
-
-        impl<Left> SelectableExpression<
-            Join<Left, $($table)::*, Inner>,
-        > for $column_name where
-            Left: $crate::JoinTo<$($table)::*>
+        impl<QS> AppearsOnTable<QS> for $column_name where
+            QS: AppearsInFromClause<$($table)::*, Count=Once>,
```

Comment:
ℹ️ Constraint A: Can only select column if column's table occurs exactly once in from/join